apps/spikeConverter.cpp

```diff
@@ -29,11 +43,46 @@ int main( int argc, char* argv[] )
     }
 
     lunchbox::Clock clock;
-    const brion::SpikeReport in( brion::URI( argv[1] ), brion::MODE_READ );
+    brion::SpikeReport in( brion::URI( argv[1] ), brion::MODE_READ );
     const float readTime = clock.resetTimef();
 
     brion::SpikeReport out( brion::URI( argv[2] ), brion::MODE_WRITE );
-    out.writeSpikes( in.getSpikes( ));
+
+    // Stream-to-File conversion
+    if( in.getReadMode() == brion::SpikeReport::STREAM )
+    {
+        while( in.getNextSpikeTime() != brion::UNDEFINED_TIMESTAMP )
+        {
+            in.waitUntil( brion::UNDEFINED_TIMESTAMP, STREAM_READ_TIMEOUT_MS );
+            out.writeSpikes( in.getSpikes( ));
+        }
+    }
+    // File-to-Stream conversion
+    else if( out.getReadMode() == brion::SpikeReport::STREAM )
+    {
+        // leave time for a Zeq connection to establish
+        lunchbox::sleep( STREAM_SEND_DELAY_MS );
+
+        brion::Spikes outSpikes;
+        unsigned int frameIndex = 0;
+        for( auto spike : in.getSpikes( ))
+        {
+            if( spike.first > (frameIndex+1) * STREAM_FRAME_LENGTH_MS )
+            {
+                out.writeSpikes( outSpikes );
+                printSentFrame( frameIndex++, outSpikes.size( ));
```

Comment:
prefix ++ we do in general but no strong opposition. The assembly code may look similar in release mode. Otherwise it is a push stack and increase operation.